docs/overview.adoc

```diff
@@ -0,0 +1,331 @@
+= Architecture
+
+== Resource types
+
+One concept this library is built around are so-called resource types.
+In the following sections we will introduce the importance of them to the overall architecture and how they are to be used to implement the SoC (separation of concerns) and dry (don't repeat yourself) principles.
+
+=== Catch up -- resource types as defined by the JSON:API specification
+
+Within the JSON:API specification every resource is assigned a type, which is simply a single string.
+The naming of the type depends on decisions made when designing the domain model of the API.
+
+If your website publishes online articles and allows users to post comments under them, an article with a single comment may exist.
+Such case is shown in the following example.
+The JSON displays a single `article` resource with the `id` 42.
+Other typical article relationships, e.g. the authors, are omitted for brevity.
+However, it contains a list of comment references, containing a single reference.
+
+[source,json]
+----
+{
+  "data": {
+    "id": "42",
+    "type": "article",
+    "attributes": {
+      "text": "…"
+    },
+    "relationships": {
+      "comments": {
+        "data": [
+          {"id": 42, "type": "comment"}
+        ]
+      }
+    }
+  }
+}
+----
+
+In this example both the article and the comment have, by chance, the same `id` field value.
+Those do however not conflict, because resources are uniquely identified by their `id` **and** `type` fields.
+
+=== Approaching the problem
+
+Above we could see that resources consist of an identifier, a type, attributes and relationships.
+All those need to be mapped to a data source, so that, for example, a `GET /article/42` request actually returns the JSON as shown above.
+
+However, it is important to note that there are different ways to get access to the data of resources:
+
+* a `GET` request to a specific identifier (e.g. `GET /article/42`) will return that specific resource
+* requests to create a resource (e.g. `POST /article/`) will return the created resource if it was not created exactly as requested
+* likewise, requests to update a resource (e.g. `PATCH /article/42`) will return the updated resource if it was not updated exactly as requested
```

Comment:
Do I understand this correctly?
In case of the resource is exactly updated as requested, nothing will be returned?
Do you have a valid example for the alternative?: The case that I want to execute an update, but it will be not be done exactly as requested? I mean isn't it dangerous to update something not exactly as requested?

Reply:
I've added links to the specification, where examples are given.